src/marvin/components/ai_function.py

```diff
@@ -124,117 +174,51 @@ def map(self, *map_args: list, **map_kwargs: list):
         each argument must be a list. The function is called once for each item
         in the list, and the results are returned in a list.
 
+        This method can be called synchronously or asynchronously.
+
         For example, fn.map([1, 2]) is equivalent to [fn(1), fn(2)].
 
-        fn.map([1, 2], x=['a', 'b']) is equivalent to [fn(1, x='a'), fn(2,
-        x='b')].
+        fn.map([1, 2], x=['a', 'b']) is equivalent to [fn(1, x='a'), fn(2, x='b')].
         """
-
-        coros = []
-
-        i = 0
-        while True:
-            call_args = []
-            call_kwargs = {}
-            try:
-                for arg in map_args:
-                    call_args.append(arg[i])
-                for k, v in map_kwargs.items():
-                    call_kwargs[k] = v[i]
-            except IndexError:
-                break
-            call_coro = self._call(*call_args, **call_kwargs)
-            coros.append(call_coro)
-            i += 1
-
-        # gather returns a future, but run_sync requires a coroutine
-        async def gather_coros():
-            return await asyncio.gather(*coros)
-
-        result = gather_coros()
-        if not self.is_async():
-            result = run_sync(result)
-        return result
-
-    async def _call(self, *args, **kwargs):
-        # Get function signature
-        sig = inspect.signature(self.fn)
-
-        # get return annotation
-        if sig.return_annotation is inspect._empty:
-            return_annotation = str
+        if asyncio.get_event_loop().is_running():
```

Comment:
I don't think we should support this pattern of unclear sync/async methods. Map is sync and Amap is async. Prefect has to support this for mixing and matching tasks but Marvin is under no obligation to introduce confusion in return signature.